pkg/sql/create_type.go

```diff
@@ -318,3 +346,21 @@ func (n *createTypeNode) Next(params runParams) (bool, error) { return false, ni
 func (n *createTypeNode) Values() tree.Datums                 { return tree.Datums{} }
 func (n *createTypeNode) Close(ctx context.Context)           {}
 func (n *createTypeNode) ReadingOwnWrites()                   {}
+
+// TODO(richardjcai): Instead of inheriting the privilege when creating the
+// descriptor, we can check the parent of type for usage privilege as well,
+// this seems to be how Postgres does it.
+// Add a test for this when we support granting privileges to schemas #50879.
+func inheritUsagePrivilegeFromSchema(
```

Comment:
I'd be a fan of having this method do the branching on what kind of schema the parent is, rather than operating on immutableschemadesc